--- ../../../../tmp/wt-2.2.4/CMakeLists.txt	2009-04-24 16:20:54.000000000 +0200
+++ CMakeLists.txt	2010-07-06 19:38:35.403071601 +0200
@@ -15,6 +15,7 @@
 SET(WTEXT_SOVERSION 7)
 SET(WTHTTP_SOVERSION 4)
 SET(WTFCGI_SOVERSION 4)
+add_definitions(-DBOOST_AUTO_LINK_NOMANGLE)
 
 #
 # Various things that must be configured by the user or packager ...
@@ -138,4 +139,4 @@
 INSTALL(DIRECTORY ${PROJECT_SOURCE_DIR}/resources DESTINATION
     ${CMAKE_INSTALL_PREFIX}/share/Wt/)
 
-INSTALL(FILES ${PROJECT_SOURCE_DIR}/wt_config.xml DESTINATION ${CONFIGDIR} )
+INSTALL(FILES ${PROJECT_SOURCE_DIR}/wt_config.xml DESTINATION ${CMAKE_INSTALL_PREFIX}/etc/wt )
