--- JSONOptions.h	2011-02-25 20:04:24.000000000 +0100
+++ libjson/JSONOptions.h	2011-05-19 11:49:38.027262608 +0200
@@ -11,7 +11,7 @@
  *  JSON_LIBRARY must be declared if libjson is compiled as a static or dynamic 
  *  library.  This exposes a C-style interface, but none of the inner workings of libjson
  */
-#define JSON_LIBRARY
+//#define JSON_LIBRARY
 
 
 /*
@@ -213,7 +213,7 @@
  *  parsing json that has comments in it as it simply ignores them, but with this option
  *  it keeps the comments and allows you to insert further comments
  */
-//#define JSON_COMMENTS
+#define JSON_COMMENTS
 
 
 /*
@@ -243,7 +243,7 @@
 /*
  *  JSON_VALIDATE turns on validation features of libjson.  This option requires JSON_SAFE
  */
-//#define JSON_VALIDATE
+#define JSON_VALIDATE
 
 
 /*
