diff -u -r ../../../../tmp/litesql-src-0.3.8/src/CMakeLists.txt ./src/CMakeLists.txt
--- ../../../../tmp/litesql-src-0.3.8/src/CMakeLists.txt	2010-06-29 11:29:42.000000000 +0200
+++ ./src/CMakeLists.txt	2010-07-14 14:20:32.063889889 +0200
@@ -1,4 +1,4 @@
 add_subdirectory(library)
-add_subdirectory(generator)
-add_subdirectory(tests)
-add_subdirectory(examples)
+#add_subdirectory(generator)
+#add_subdirectory(tests)
+#add_subdirectory(examples)
diff -u -r ../../../../tmp/litesql-src-0.3.8/src/examples/CMakeLists.txt ./src/examples/CMakeLists.txt
--- ../../../../tmp/litesql-src-0.3.8/src/examples/CMakeLists.txt	2010-06-29 11:29:42.000000000 +0200
+++ ./src/examples/CMakeLists.txt	2010-07-14 14:20:32.064889922 +0200
@@ -21,7 +21,7 @@
  
 add_executable(example ${example_SOURCES} ${example_HEADERS})
 
-target_link_libraries (example litesql litesql-util ${BACKEND_LIBRARIES})
+target_link_libraries (example litesql litesql-util ${BACKEND_LIBRARIES} crypt z pthread)
 
 set_property(TARGET example APPEND PROPERTY COMPILE_DEFINITIONS ${SQLITE3_DEFINITIONS})
 
diff -u -r ../../../../tmp/litesql-src-0.3.8/src/library/CMakeLists.txt ./src/library/CMakeLists.txt
--- ../../../../tmp/litesql-src-0.3.8/src/library/CMakeLists.txt	2010-06-29 11:29:47.000000000 +0200
+++ ./src/library/CMakeLists.txt	2010-07-14 14:31:59.945890173 +0200
@@ -47,10 +47,10 @@
 			COMPONENT devel)
 install(EXPORT ${LIBNAME} DESTINATION lib)
 
-if (MSVC)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/debug/litesqld.lib DESTINATION lib/static
-	COMPONENT devel)
-endif(MSVC)
+#if (MSVC)
+#install(FILES ${CMAKE_CURRENT_BINARY_DIR}/debug/litesqld.lib DESTINATION lib/static
+#	COMPONENT devel)
+#endif(MSVC)
 
 if (WITH_SQLITE3)
   set(${LIBNAME}_sqlite_SRC
@@ -71,9 +71,9 @@
 			ARCHIVE DESTINATION lib/static 
 			COMPONENT devel)
 install(EXPORT ${LIBNAME}_sqlite DESTINATION lib)
-if (MSVC)
-  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/debug/${LIBNAME}_sqlited.lib DESTINATION lib/static COMPONENT devel)
-endif(MSVC)
+#if (MSVC)
+#  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/debug/${LIBNAME}_sqlited.lib DESTINATION lib/static COMPONENT devel)
+#endif(MSVC)
 endif (WITH_SQLITE3)
 
 if (WITH_MYSQL)
@@ -81,14 +81,15 @@
         				mysql.cpp
    
   )
+
   install(TARGETS ${LIBNAME}_mysql  EXPORT ${LIBNAME}_mysql 
 			RUNTIME DESTINATION lib 
 			ARCHIVE DESTINATION lib/static 
 			COMPONENT devel)
   install(EXPORT ${LIBNAME}_mysql DESTINATION lib)
-if (MSVC)
-  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/debug/${LIBNAME}_mysqld.lib DESTINATION lib/static COMPONENT devel)
-endif(MSVC)
+#if (MSVC)
+#  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/debug/${LIBNAME}_mysqld.lib DESTINATION lib/static COMPONENT devel)
+#endif(MSVC)
 
 endif (WITH_MYSQL)
 
@@ -100,9 +101,9 @@
 			RUNTIME DESTINATION lib 
 			ARCHIVE DESTINATION lib/static 
 			COMPONENT devel)
-if (MSVC)
-  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/debug/${LIBNAME}_postgresqld.lib DESTINATION lib/static COMPONENT devel)
-endif(MSVC)
+#if (MSVC)
+#  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/debug/${LIBNAME}_postgresqld.lib DESTINATION lib/static COMPONENT devel)
+#endif(MSVC)
 endif (WITH_POSTGRES)
 
 if (WITH_ODBC)
@@ -114,9 +115,9 @@
 			RUNTIME DESTINATION lib 
 			ARCHIVE DESTINATION lib/static 
 			COMPONENT devel)
-if (MSVC)
-  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/debug/${LIBNAME}_odbcd.lib DESTINATION lib/static COMPONENT devel)
-endif(MSVC)
+#if (MSVC)
+#  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/debug/${LIBNAME}_odbcd.lib DESTINATION lib/static COMPONENT devel)
+#endif(MSVC)
 
 endif (WITH_ODBC)
 
diff -u -r ../../../../tmp/litesql-src-0.3.8/src/library/mysql.cpp ./src/library/mysql.cpp
--- ../../../../tmp/litesql-src-0.3.8/src/library/mysql.cpp	2010-06-29 11:29:47.000000000 +0200
+++ ./src/library/mysql.cpp	2010-07-14 14:20:32.064889922 +0200
@@ -8,7 +8,9 @@
 #ifdef HAVE_LIBMYSQLCLIENT
 #include <string>
 
+#ifdef WIN32
 #include "config-win.h"
+#endif
 #include <mysql.h>
 
 using namespace litesql;
diff -u -r ../../../../tmp/litesql-src-0.3.8/src/tests/CMakeLists.txt ./src/tests/CMakeLists.txt
--- ../../../../tmp/litesql-src-0.3.8/src/tests/CMakeLists.txt	2010-06-29 11:29:48.000000000 +0200
+++ ./src/tests/CMakeLists.txt	2010-07-14 14:20:32.064889922 +0200
@@ -11,7 +11,7 @@
 add_test(test-datetime ${EXECUTABLE_OUTPUT_PATH}/test-datetime datetime)
 
 add_executable(test-update-table test-update-table.cpp)
-target_link_libraries (test-update-table litesql litesql-util ${BACKEND_LIBRARIES})
+target_link_libraries (test-update-table litesql litesql-util ${BACKEND_LIBRARIES} crypt z pthread)
 add_test(test-update-table ${EXECUTABLE_OUTPUT_PATH}/test-update-table update-table)
 
 INCLUDE_DIRECTORIES(../library)
